src/components/DonateMoney.vue

```diff
@@ -25,22 +83,75 @@
 <script lang="js">
 import axios from 'axios'
 export default {
-    name: 'DonateMoney',
-    props: [],
-    data () {
-      return {
-        donation: 1.50
-      }
+  name: 'DonateMoney',
+  props: [],
+  data () {
+    return {
+      donationAmount: 2,
+      customAmountSelected: false,
+      customDonationAmount: undefined,
+      message: ''
+    }
+  },
+  computed: {
+  },
+  mounted () {
+  },
+  methods: {
+    unsetCustomAmountSelection() {
+      this.customAmountSelected = false;
+      this.message = "";
     },
-    computed: {
+    handleCustomAmountSelection() {
+      this.customAmountSelected = !this.customAmountSelected;
+      this.message = "";
     },
-    mounted () {
+    validateDonationAmount(value = this.customDonationAmount) {
+      let isValid = true;
+      value = parseFloat(value); // outputs: number
+      value = value.toFixed(2); // outputs: string
+      value = parseFloat(value); // outputs: number
+      // separating parameter assignment and parseFloat operation for consistent outcome on change
+
+      // if an existing donation amount button was chosen
+      if (!this.customAmountSelected) {
+        value = this.donationAmount;
+        return { value, isValid } // i.e. { 2 || 20 || 50, true }
+      }
+
+      if (value > 1.49 && value < 10000.01) {
+        this.message = "";
+        return { value, isValid }; // { 1.50, true }
+      }
+
+      if (isNaN(value)) { // undefined || NaN
+        this.message = "Please select or enter a valid amount.";
+      }
+      if (value < 1.5) {
+        this.message = "Please enter a donation amount higher than $1.50.";
+      }
+      if (value > 10000) {
+        this.message = "Amplify currently only accept donation amounts less than $10,000."
+      }
+
+      isValid = false;
+      return { value, isValid }; // i.e. { undefined, false }
```

Comment:
This works, but seems unnecessarily complicated. The method name implies that the function is just doing validation, but there is some other logic happening which is confusing. Consider splitting this into a pure validation method and a method to handle whether or not the custom donation is being used.

Reply:
Hi Glenn, thanks for the feedback. 

In my mind, what the validate function should return is a boolean and the acceptable value. I figured the user feedback is important as well. As I'm trying to understand which part is unnecessarily complicated, and I'm guessing perhaps you are referring to how I am additionally checking if custom amount is selected inside of `validateDonationAmount`, so lines 116-120:

https://github.com/ProgramEquity/amplify/blob/c0b1cb989b29e834c8a98b6a098d325128f930c1/src/components/DonateMoney.vue#L116-L120

A few things about this: 
(1) Do you think it's necessary to validate a value that is preselected? One approach can be that I can remove lines 116-120 because, arguably, a pre-selected value should not need to be validated. I don't know if there is (enough of) a security issue here. Removing those lines alone, Does this make the function a pure validation method?
(2) Or, perhaps you want me to separate the logic to provide user feedback from the validate function? Or did you want me to return only a boolean for this validate function? If the function returns a boolean, then somewhere else the number will need to be sanitized though.
(3) I checked out the `SignName` component and learned a few things about using `rules` and `ref`. As it currently is in `SignName`, I don't think it works though. `rules` do not validate nor does it sanitize the input though, although it provides a consistent visual for providing visual feedback to the user. I can try that implementation of the same feature. Would you like me to do that instead?

I can think of a few different ways to change my approach, but would love to get clarity on scope and expectations. Thanks!

Reply:
I've created a `format` function, a `validate` function and saved them in `util` folder so that both frontend and backend can use them. In addition, although I liked my custom solution for providing user feedback on custom donation amount input, I'm going to use the `rules` property to invoke custom functions based on `v-text-field`'s state for some visual consistency. (ref: [validation & rules](https://vuetifyjs.com/en/components/text-fields/#validation-26-rules)) 

https://github.com/ProgramEquity/amplify/blob/ac3c112098522642d87d6c0167ef48285a3c0aff/src/components/DonateMoney.vue#L93

https://github.com/ProgramEquity/amplify/blob/ac3c112098522642d87d6c0167ef48285a3c0aff/src/components/DonateMoney.vue#L114